### PR TITLE
Improve memory usage and detection for HashJoin iterator

### PR DESCRIPTION
**Pull request by matriv**
see commit messages

### CODE REVIEW

**Comment by andreidan** [comment]
We need to double check if the 184 overhead / key is correct.

**Comment by andreidan** [comment]
Also check if ObjectArrayList is better from a memory usage perspective than a LinkedList

**Comment by matriv** [comment]
@andreidan Changed to plain java ArrayList and updated overhead calculation.
CircuitBreaker is thrown for 256m of heap and for 512m query returns (uservisits dataset)

File: sql/src/main/java/io/crate/breaker/RowAccounting.java [hunk 1]
```diff
@@ -41,6 +42,13 @@ public RowAccounting(Collection<? extends DataType> columnTypes, RamAccountingCo
         this.ramAccountingContext = ramAccountingContext;
     }
 
+    public RowAccounting(Collection<? extends DataType> columnTypes,
+                         RamAccountingContext ramAccountingContext,
+                         int extraSizePerRow) {
```
Comment (andreidan) [inline sql/src/main/java/io/crate/breaker/RowAccounting.java]: can you please provide some more context around what `extraSizePerRow` is (maybe even the array example). being a public constructor I think it'd be helpfull

**Pull request merged**